packages/app/src/ChartUtils.tsx

```diff
@@ -919,3 +938,198 @@ export const convertV1ChartConfigToV2 = (
+  }
+
+  const additionalFilters: Filter[] = [];
+
+  // Add group-by column filters
+  if (groupFilters && groupFilters.length > 0) {
+    groupFilters.forEach(({ column, value }) => {
+      if (column && value != null) {
+        // Can't use SQLString.escape here because the search endpoint relies on exist match for UI
+        const condition = `${column} IN (${SqlString.escape(value)})`;
+        additionalFilters.push({ type: 'sql', condition });
+      }
+    });
+  }
+
+  // Add Y-axis value range filter (±threshold) for charts
+  if (valueRangeFilter) {
+    const { expression, value, threshold = 0.05 } = valueRangeFilter;
+    const hasAggregateFunction = isAggregateFunction(expression);
+
+    if (!hasAggregateFunction) {
+      const lowerBound = value * (1 - threshold);
+      const upperBound = value * (1 + threshold);
+      // Can't use SQLString.escape here because the search endpoint relies on exist match for UI
+      const condition = `${expression} BETWEEN ${SqlString.escape(lowerBound)} AND ${SqlString.escape(upperBound)}`;
+
+      additionalFilters.push({
+        type: 'sql',
+        condition,
+      });
+    }
+  }
+
+  // Get the time range
+  const from = dateRange[0].getTime();
+  const to = dateRange[1].getTime();
+
+  const params: Record<string, string> = {
+    source: source?.id ?? '',
+    where: where,
+    whereLanguage: whereLanguage,
+    filters: JSON.stringify([...(config.filters ?? []), ...additionalFilters]),
+    isLive: 'false',
+    from: from.toString(),
+    to: to.toString(),
+  };
+
+  // If its a metric chart, we don't can't pass the where and filters
+  if (isMetricChart) {
+    params.where = '';
+    params.whereLanguage = 'lucene';
+    params.filters = JSON.stringify([]);
+    params.source = source?.logSourceId ?? '';
+  }
+
+  // Include the select parameter if provided to preserve custom columns
+  // eventTableSelect is used for charts that override select (like histograms with count)
+  // to preserve the original table's select expression
+  if (config.eventTableSelect) {
+    params.select = config.eventTableSelect;
+  }
+
+  return `/search?${new URLSearchParams(params).toString()}`;
+}
+
+/**
+ * Extract group column names from chart config's groupBy field
+ * Handles both string format ("col1, col2") and array format ([{ valueExpression: "col1" }, ...])
+ */
+function extractGroupColumns(
+  groupBy: ChartConfigWithDateRange['groupBy'],
+): string[] {
+  if (!groupBy) return [];
+
+  if (typeof groupBy === 'string') {
+    // String GROUP BY: "col1, col2"
+    return groupBy.split(',').map(v => v.trim());
+  }
+
+  // Array GROUP BY: [{ valueExpression: "col1" }, ...] or ["col1", ...]
+  return groupBy.map(g => (typeof g === 'string' ? g : g.valueExpression));
+}
+
+/**
+ * Build search URL from a table row click
+ * Extracts group filters and value range filter from the row data
+ */
+export function buildTableRowSearchUrl({
+  row,
+  source,
+  config,
```

Comment:
Not for now since this implementation is complete, but would it be possible to instead generate URLs from the return data instead of the config? In theory - this will allow us to take any response from ClickHouse (including raw user-defined SQL queries) and still behave properly for table responses (which is why some other chart util functions use return type information as opposed to chart config information)

This would also allow us to take advantage of logic in things like https://github.com/hyperdxio/hyperdx/blob/586bcce7f1bf214f5f84be10f53f869ecd89af8c/packages/app/src/ChartUtils.tsx#L516